### PR TITLE
API-5999: Fortify scan April 2021

### PR DESCRIPTION
**Pull request by jabando-va**

### CODE REVIEW

File: charon-models/src/main/java/gov/va/api/lighthouse/charon/models/vprgetpatientdata/VprGetPatientData.java [hunk 1]
```diff
@@ -109,8 +109,8 @@ public RpcDetails asDetails() {
                   RpcDetails.Parameter.builder()
                       .string(
                           type().stream()
-                              .filter(Objects::nonNull)
-                              .map(Enum::name)
+                              .filter(m -> Objects.nonNull(m))
```
Comment (bschofield-va) [inline charon-models/src/main/java/gov/va/api/lighthouse/charon/models/vprgetpatientdata/VprGetPatientData.java:112]: no, why are we removing the method reference...

File: charon/src/main/java/gov/va/api/lighthouse/charon/service/controller/ApplicationProxyUserVistalinkSession.java [hunk 2]
```diff
@@ -61,7 +62,7 @@ private void closeConnection() {
     }
     try {
       connection.close();
-    } catch (Exception e) {
+    } catch (ResourceException e) {
```
Comment (bschofield-va) [inline charon/src/main/java/gov/va/api/lighthouse/charon/service/controller/ApplicationProxyUserVistalinkSession.java:65]: I don't like this either... we don't know what awesome exceptions vistalink libraries can throw.

File: charon/src/main/java/gov/va/api/lighthouse/charon/service/controller/StandardUserVistalinkSession.java [hunk 3]
```diff
@@ -69,7 +70,7 @@ private void closeConnection() {
     }
     try {
       connection.close();
-    } catch (Exception e) {
+    } catch (ResourceException e) {
```
Comment (bschofield-va) [inline charon/src/main/java/gov/va/api/lighthouse/charon/service/controller/StandardUserVistalinkSession.java:73]: `#same`

File: charon-models/src/main/java/gov/va/api/lighthouse/charon/models/FilemanDate.java [hunk 4]
```diff
@@ -65,7 +65,8 @@ public BadFilemanDate(String reason, char[] value) {
   }
 
   private static class DateTimeFormatter {
```
Comment (bschofield-va) [inline charon-models/src/main/java/gov/va/api/lighthouse/charon/models/FilemanDate.java:67]: This class isn't thread safe, nor does it need to be. It's created/used/discarded in one statement.
I'm not going to die on this hill, but this is unnecessary.

File: charon/src/main/java/gov/va/api/lighthouse/charon/service/controller/ApplicationProxyUserVistalinkSession.java (hunk 2 above)
Comment (jabando-va) [reply charon/src/main/java/gov/va/api/lighthouse/charon/service/controller/ApplicationProxyUserVistalinkSession.java:65]: The issue on this was using generic Exceptions, I tried using more specific ones when possible; like this one. Ill just revert these and then we should develop a write off referring to the vistalink libraries for the next scan.

**Review by jhulbert-va — COMMENTED** [review]
@bschofield-va I talked through some of these with JJ, so if you're good with em, I think it's good to merge.

**Review by bschofield-va — APPROVED** [review]

**Pull request merged**